### PR TITLE
Update SFT trainer test script to run on Airflow

### PR DESCRIPTION
**Pull request by SurbhiJainUSC**
# Description
This PR updates the end-to-end test script for SFT Trainer that will run on Airflow DAG. The test script will do the following:
1. Run fine-tuning using SFT.
2. Run inference on a test prompt.
3. Run SFT trainer correctness test to match the logits.

*NOTE*: This PR depends on [HF Tokenizer Support PR](https://github.com/AI-Hypercomputer/maxtext/pull/1439). Merge this only when HF Tokenizer PR is merged, otherwise the airflow tests would fail.

*Notice 1:* Once all tests pass, the "pull ready" label will automatically be assigned.
This label is used for administrative purposes. Please do not add it manually.

*Notice 2:* For external contributions, our settings currently require an approval from a MaxText maintainer to trigger CI tests.

# Tests
Ran airflow test on local server. Related airflow test PR: 

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

**Review by bvandermoon — APPROVED** [review]

**Review by RissyRan — APPROVED** [review]
LGTM
